### PR TITLE
Remove connection type constraint

### PR DESCRIPTION
**Pull request by jstrutz**
## Description
Remove expectation that connection type string will be a previously-known member of an enum.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: src/main/kotlin/com/workos/sso/SsoApi.kt [hunk 1]
```diff
@@ -26,7 +26,7 @@ class SsoApi(private val workos: WorkOS) {
    * @param redirectUri A redirect URI to return an authorized user to.
    * @param connection Connection ID to determine which identity provider to authenticate with.
    * @param domain Use the domain to determine which connection and identity provider to authenticate with.
-   * @param provider Name of the identity provider e.g. GoogleOAuth or MicrosoftOAuth.
+   * @param provider Name of the identity provider e.g. GithubOAuth, GoogleOAuth, or MicrosoftOAuth.
```
Comment (oliverzheng) [inline src/main/kotlin/com/workos/sso/SsoApi.kt:29]:
```suggestion
   * @param provider Name of the identity provider e.g. GitHubOAuth, GoogleOAuth, or MicrosoftOAuth.
```

File: src/main/kotlin/com/workos/sso/SsoApi.kt [hunk 2]
```diff
@@ -78,7 +78,7 @@ class SsoApi(private val workos: WorkOS) {
     /**
      * Value used to authenticate all users with the same connection and Identity Provider.
      *
-     * Currently, the only supported values for provider are GoogleOAuth and
+     * Currently, the only supported values for provider are GithubOAuth, GoogleOAuth, and
```
Comment (oliverzheng) [inline src/main/kotlin/com/workos/sso/SsoApi.kt:81]:
```suggestion
     * Currently, the only supported values for provider are GitHubOAuth, GoogleOAuth, and
```

File: src/main/kotlin/com/workos/sso/models/ConnectionType.kt
Comment (oliverzheng) [inline src/main/kotlin/com/workos/sso/models/ConnectionType.kt:1]: I imagine the build failures are all around places that reference this `ConnectionType` still.

File: build.gradle.kts [hunk 3]
```diff
@@ -108,7 +108,7 @@ tasks.jar {
 
 java {
   toolchain {
-    languageVersion.set(JavaLanguageVersion.of(8))
```
Comment (oliverzheng) [inline build.gradle.kts:111]: Is this change intended?

**Review by oliverzheng — COMMENTED** [review]
Looks good! Just 1 question.

File: build.gradle.kts (hunk 3 above)
Comment (jstrutz) [reply build.gradle.kts:111]: Oops -- no, that was to get it to build on my machine.  I will remove.

**Review by oliverzheng — APPROVED** [review]
